### PR TITLE
Allow empty responses for subscriptions

### PR DESCRIPTION
**Pull request by dragid10**
Should fix https://github.com/allenporter/python-google-nest-sdm/issues/1134

This change should fix an exception when we're unable to find subscriptions. By using the dict's `get_or_default` method, we can still just return an empty list instead of erroring out completely.

The error looks like this:
```
  File "/usr/src/homeassistant/homeassistant/components/nest/config_flow.py", line 298, in async_step_pubsub_topic
    return await self.async_step_pubsub_subscription()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/nest/config_flow.py", line 368, in async_step_pubsub_subscription
    await self._admin_client.list_eligible_subscriptions(
  File "/usr/local/lib/python3.12/site-packages/google_nest_sdm/admin_client.py", line 230, in list_eligible_subscriptions
    subscriptions = await self.list_subscriptions(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/google_nest_sdm/admin_client.py", line 168, in list_subscriptions
    return response["subscriptions"]  # type: ignore[no-any-return]
           ~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'subscriptions'
```

### CODE REVIEW

**Review by allenporter — CHANGES_REQUESTED** [review]
Appreciate the fix! Please revert the formatting changes and send in another PR.

File: tests/test_admin_client.py [hunk 1]
```diff
@@ -307,6 +307,30 @@ async def test_list_subscriptions(
...
+async def test_list_subscriptions_empty_response(
+    app: aiohttp.web.Application,
+    admin_client: Callable[[], Awaitable[AdminClient]],
+    recorder: Recorder,
+) -> None:
+    """Test listing subscriptions."""
+
+    handler = NewHandler(
+        recorder,
+        [{"subscriptions": []}],
```
Comment (allenporter) [inline tests/test_admin_client.py:319]: Is this actually exercising the bug? I think the bug is when the response is `{}`

**Review by allenporter — CHANGES_REQUESTED** [review]

File: tests/test_admin_client.py (hunk 1 above)
Comment (allenporter) [inline tests/test_admin_client.py:319]: I'm pretty sure its not. (Aside: Usually a best practice is to write the test first, ensure its failing, then add the fix that makes the test pass)

```suggestion
        [{}]
```

**Review by allenporter — CHANGES_REQUESTED** [review]

File: tests/test_admin_client.py (hunk 1 above)
Comment (dragid10) [reply tests/test_admin_client.py:319]: Ahh yep you're right! 


The original error is a `KeyError` with the `subscriptions` key missing from the response:
```python
return response["subscriptions"]
```

Meaning that the `response` json could either:
1. Have other properties except `subscriptions` in it
or
2. Be an empty json

Thanks for checking my logic on that 🙌🏿
I'll go ahead and commit that change

**Review by allenporter — APPROVED** [review]
Thank you for the fix @dragid10 👍🏼

**Pull request merged**